homeassistant/components/matrix/__init__.py

```diff
@@ -255,10 +297,25 @@ async def _handle_room_message(self, room: MatrixRoom, message: Event) -> None:
                 "command": command[CONF_NAME],
                 "sender": message.sender,
                 "room": room_id,
+                "event_id": message.event_id,
                 "args": match.groupdict(),
             }
+            if thread_parent := self._get_thread_parent(message):
+                message_data["thread_parent"] = thread_parent
             self.hass.bus.async_fire(EVENT_MATRIX_COMMAND, message_data)
 
+    def _get_thread_parent(self, message: RoomMessageText) -> Any | None:
+        """Get the thread parent ID from a message, or None if not in a thread."""
+        content: dict[str, Any] | None = message.source.get("content")
+        if content is not None:
+            if (relates_to := content.get("m.relates_to")) is not None:
+                if (
+                    relates_to.get("rel_type") == "m.thread"
+                    and (event_id := relates_to.get("event_id")) is not None
+                ):
+                    return event_id
+        return None
```

Comment:
```suggestion
        match message.source:
            case {"content": {"m.relates_to": {"rel_type": "m.thread", "event_id": str() as event_id}}}:
                return event_id
            case _:
                return None
```

Reply:
Oh - nice, I didn't realize that Python had match expressions. I don't frequently use Python. Thanks - this is much clearer.